samples/sid_end_device/src/sidewalk.c

```diff
@@ -316,31 +317,26 @@ static void state_sidewalk_run(void *o)
 	} break;
 	case SID_EVENT_FILE_TRANSFER: {
 #ifdef CONFIG_SIDEWALK_FILE_TRANSFER
+		static struct flash_img_context img_ctx;
 		struct data_received_args *args = (struct data_received_args *)sm->event.ctx;
 		if (!args) {
 			LOG_ERR("File transfer event data is NULL");
 			break;
 		}
 		LOG_INF("Received file Id %d; buffer size %d; file offset %d", args->desc.file_id,
 			args->buffer->size, args->desc.file_offset);
-		uint8_t hash_out[32];
-		sid_pal_hash_params_t params = { .algo = SID_PAL_HASH_SHA256,
-						 .data = args->buffer->data,
-						 .data_size = args->buffer->size,
-						 .digest = hash_out,
-						 .digest_size = sizeof(hash_out) };
-
-		sid_error_t e = sid_pal_crypto_hash(&params);
-		if (e != SID_ERROR_NONE) {
-			LOG_ERR("Failed to hash received file transfer with error %s",
-				SID_ERROR_T_STR(e));
-		} else {
-#define HEX_PRINTER(a, ...) "%02X"
-#define HEX_PRINTER_ARG(a, ...) hash_out[a]
-			char hex_str[sizeof(hash_out) * 2 + 1] = { 0 };
-			snprintf(hex_str, sizeof(hex_str), LISTIFY(32, HEX_PRINTER, ()),
-				 LISTIFY(32, HEX_PRINTER_ARG, (, )));
-			LOG_INF("SHA256: %s", hex_str);
```

Comment:
Our CI tests rely on the hash output of the payload so we shouldn't delete this (at liest not without changes made in tests)

 would propose to extract this to a function (to help with swapping this implementation to new one) and contitionally call the extracted function ( or the new implementation) based on the KConfig feature flag setting. 
 
 I would propose to set the default value of this feature flag to use the old implementation.
 
 Not everything need to be behind the feature flag setting. The changes in the structures, additional fields does not impact previous implementation, so they can be added as is.

Reply:
thanks for pointing this out, will be reversed before converting draft to pr